src/resolvers/exotics/file-resolver.js

```diff
@@ -23,6 +24,11 @@ export default class FileResolver extends ExoticResolver {
   loc: string;
 
   static protocol = 'file';
+  static prefixMatcher = /^.{1,2}\//;
```

Comment:
Oops, looks like the `.` isn't escaped, so it's matching _any_ two characters, including single-character scopes like `@m`.

I have a fix here: https://github.com/yarnpkg/yarn/pull/4338